### ISSUE TITLE
自建推理脚本与训练时的推理无法对齐

### ISSUE THREAD

**Issue by ICE-Bruce** [hiyouga/EasyR1#499]
您好，感谢您的工作。我使用了qwen自带的推理脚本，但是结果发现无法与训练中的测试集推理结果对齐，请问有哪里需要修改吗，谢谢：
下述输入的prompt已经提前添加了./examples/format_prompt/math.jinja内容。

`import os
import json
import socket # Used for dynamic port selection
import torch # Import torch

from transformers import Qwen2_5_VLForConditionalGeneration, AutoProcessor
from qwen_vl_utils import process_vision_info 

# --- 配置部分 ---
INPUT_JSONL_PATH = ""
OUTPUT_JSONL_PATH=""
# --- 模型和处理器加载 (保持不变) ---
print("Loading model and processor...")
model_checkpoint_path=""

model = Qwen2_5_VLForConditionalGeneration.from_pretrained(
    model_checkpoint_path, torch_dtype="auto", device_map="auto")

processor = AutoProcessor.from_pretrained("Qwen2.5-VL-7B-Instruct")
print("Model and processor loaded.")

# --- 处理jsonl文件 ---
processed_count = 0
skipped_count = 0

with open(INPUT_JSONL_PATH, 'r', encoding='utf-8') as infile, \
     open(OUTPUT_JSONL_PATH, 'w', encoding='utf-8') as outfile:
    
    for line in infile:
        try:
            data = json.loads(line.strip())
            
            img = data.get("image")
            conversations = data.get("conversations")
            prompt_text = data.get("prompt")
            # 构建messages输入格式
            messages = [
                {
                    "role": "user",
                    "content": [
                        {"type": "image", "image": img},
                        {"type": "text", "text": prompt_text.replace('<image>', '').replace('</image>', '')},
                    ],
                }
            ]

            # 准备推理输入
            text = processor.apply_chat_template(
                messages, tokenize=False, add_generation_prompt=True
            )
            print("text:", text)
            image_inputs, video_inputs = process_vision_info(messages) # 这里的messages必须和chat_template里用的保持一致
            
            inputs = processor(
                text=[text],
                images=image_inputs,
                videos=video_inputs,
                padding=True,
                return_tensors="pt",
            )
            inputs = inputs.to(model.device)

            # 推理
            generated_ids = model.generate(**inputs, max_new_tokens=2000)
            generated_ids_trimmed = [
                out_ids[len(in_ids) :] for in_ids, out_ids in zip(inputs.input_ids, generated_ids)
            ]
            output_text = processor.batch_decode(
                generated_ids_trimmed, skip_special_tokens=True, clean_up_tokenization_spaces=False
            )
            
            # 将模型推理结果添加到原始json数据中
            data["qwen_vl_prediction"] = output_text[0].strip() if output_text else ""
            print(f"  Prediction: {data['qwen_vl_prediction']}")
            
            # 将更新后的json写入输出文件
            outfile.write(json.dumps(data, ensure_ascii=False) + '\n')
            processed_count += 1

        except json.JSONDecodeError as e:
            print(f"Error decoding JSON: {e} in line: {line.strip()}")
            skipped_count += 1
        except Exception as e:
            print(f"An error occurred during processing line: {line.strip()}\nError: {e}")
            skipped_count += 1
            # 可以在这里选择是否将错误信息添加到数据中，并写入文件
            try:
                err_data = json.loads(line.strip())
                err_data["qwen_vl_prediction"] = f"ERROR: {str(e)}"
                outfile.write(json.dumps(err_data, ensure_ascii=False) + '\n')
            except:
                pass # 如果连原始行都无法解析，则跳过写入

print(f"\nProcessing complete. Total processed: {processed_count}, Total skipped: {skipped_count}")
print(f"Results saved to {OUTPUT_JSONL_PATH}")

`

**Comment by hiyouga** [comment]
Please check if the input ids and pixel values are the same.
